### PR TITLE
Add caching to profanity filter route

### PR DESCRIPTION
**Pull request by JillianK**
This adds caching to the profanity filter route. Currently, we check webpurify each time a student uses a project with playSpeech in it for the first time. This will cache those results so we don't make as many webpurify requests for the same text.

## Links

- [slack thread about this](https://codedotorg.slack.com/archives/C0T10HG6N/p1593731050038200)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/profanity_controller.rb [hunk 1]
```diff
@@ -4,8 +4,18 @@ class ProfanityController < ApplicationController
...
   def find
-    render json: ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
+    profanity_result = []
+    if params[:should_cache]
+      cache_key = "profanity/#{params[:language] || request.locale}/#{params[:text] || ''}"
+      profanity_result = Rails.cache.fetch(cache_key) do
+        ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
+      end
+    else
+      profanity_result = ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
```
Comment (breville) [inline dashboard/app/controllers/profanity_controller.rb]: I guess this question applies to existing code too, but is there any point calling `find_potential_profanities` if `params[:text]` is `nil`?

Comment (JillianK) [reply dashboard/app/controllers/profanity_controller.rb]: not really, I'll add a check for it

File: dashboard/app/controllers/profanity_controller.rb [hunk 2]
```diff
@@ -8,13 +8,15 @@ class ProfanityController < ApplicationController
   # @returns [Array<String>|nil] Profane words within the given string
   def find
     profanity_result = []
-    if params[:should_cache]
-      cache_key = "profanity/#{params[:language] || request.locale}/#{params[:text] || ''}"
-      profanity_result = Rails.cache.fetch(cache_key) do
-        ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
+    unless params[:text].nil?
+      if params[:should_cache]
+        cache_key = "profanity/#{params[:language] || request.locale}/#{params[:text] || ''}"
```
Comment (breville) [inline dashboard/app/controllers/profanity_controller.rb]: nit: do we still need the `|| ''` here?

File: dashboard/app/controllers/profanity_controller.rb [hunk 3]
```diff
@@ -8,13 +8,15 @@ class ProfanityController < ApplicationController
...
+        profanity_result = Rails.cache.fetch(cache_key) do
```
Comment (breville) [inline dashboard/app/controllers/profanity_controller.rb:14]: Is it okay to have any form of user-submitted text as our cache key?  Any issues with emoji etc.?

Comment (JillianK) [reply dashboard/app/controllers/profanity_controller.rb:14]: it should be fine, I've tested it with a bunch of alphabets and it just reads the emoji description when you provide that.

File: dashboard/app/controllers/profanity_controller.rb (hunk 2 above)
Comment (JillianK) [reply dashboard/app/controllers/profanity_controller.rb]: oops nope

**Review by breville — APPROVED** [review]
Great work! 🤬

**Pull request merged**